src/Servers/Kestrel/Transport.MsQuic/src/Internal/MsQuicNativeMethods.cs

```diff
@@ -536,5 +494,105 @@ internal struct SOCKADDR_INET
             [FieldOffset(0)]
             internal ushort si_family;
         }
+
+        internal delegate uint StreamCallbackDelegate(
+            IntPtr Stream,
+            IntPtr Context,
+            ref StreamEvent Event);
+
+        internal delegate uint StreamOpenDelegate(
+            IntPtr Connection,
+            uint Flags,
+            StreamCallbackDelegate Handler,
+            IntPtr Context,
+            out IntPtr Stream);
+
+        internal delegate uint StreamStartDelegate(
+            IntPtr Stream,
+            uint Flags
+            );
+
+        internal delegate uint StreamCloseDelegate(
+            IntPtr Stream);
+
+        internal delegate uint StreamShutdownDelegate(
+            IntPtr Stream,
+            uint Flags,
+            ushort ErrorCode);
+
+        internal delegate uint StreamSendDelegate(
+            IntPtr Stream,
+            QuicBuffer* Buffers,
+            uint BufferCount,
+            uint Flags,
+            IntPtr ClientSendContext);
+
+        internal delegate uint StreamReceiveCompleteDelegate(
+            IntPtr Stream,
+            ulong BufferLength);
+
+        [StructLayout(LayoutKind.Sequential)]
+        internal unsafe struct QuicBuffer
+        {
+            internal uint Length;
+            internal byte* Buffer;
+        }
+
+        public static SOCKADDR_INET Convert(IPEndPoint endpoint)
+        {
+            var socketAddress = new SOCKADDR_INET();
+            var buffer = endpoint.Address.GetAddressBytes();
+            switch (endpoint.Address.AddressFamily)
+            {
+                case AddressFamily.InterNetwork:
+                    socketAddress.Ipv4.sin_addr0 = buffer[0];
+                    socketAddress.Ipv4.sin_addr1 = buffer[1];
+                    socketAddress.Ipv4.sin_addr2 = buffer[2];
+                    socketAddress.Ipv4.sin_addr3 = buffer[3];
+                    socketAddress.Ipv4.sin_family = 0;
```

Comment:
Shouldn't this be set?

Reply:
It was set to zero before. This means that MsQuic will listen for both IPv4 and IPv6 for the same port. Not sure if we should allow this to be configurable or not.

Reply:
You do need to set the family. Yes, msquic supports listening to AF_UNSPEC, but if the endpoint is a v4 or v6 you should set the family accordingly.

Reply:
Unspecified only makes sense when you're not setting an IP.

Reply:
MsQuic supports listening on unspecified address, but a specified port.

Reply:
> Unspecified only makes sense when you're not setting an IP.

So listening to any address (0.0.0.0) on a certain port?

Reply:
Yeah, I think this code only works right now because the sample is passing in 0.0.0.0. If you put in a specific IP without setting the address family it probably won't work.